pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [insert violates foreign key constraint]

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://mesolite.cluster.engflow.com/invocations/default/88dc33b4-2377-400d-ae58-2a8edaf374ef?testReportRun=5&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvd29ya2xvYWQvc2NoZW1hY2hhbmdlOnNjaGVtYWNoYW5nZV90ZXN0) on release-24.3.21-rc @ [d343b53a21d34412715ed228d74ca4f114cf147c](https://github.com/cockroachdb/cockroach/commits/d343b53a21d34412715ed228d74ca4f114cf147c):


```
   },
   {
    "query": "SELECT ARRAY[parent.table_schema, parent.table_name, parent.column_name, child.column_name] FROM (SELECT conname, conkey, confkey, conrelid, confrelid FROM pg_constraint WHERE (contype = 'f') AND (conrelid = 'public.table_w1_124'::REGCLASS::INT8)) AS con JOIN (SELECT column_name, ordinal_position, column_default FROM information_schema.columns WHERE (table_schema = 'public') AND (table_name = 'table_w1_124')) AS child ON conkey[1] = child.ordinal_position JOIN (SELECT pc.oid, cols.table_schema, cols.table_name, cols.column_name, cols.ordinal_position FROM pg_class AS pc JOIN pg_namespace AS pn ON pc.relnamespace = pn.oid JOIN information_schema.columns AS cols ON ((pc.relname = cols.table_name) AND (pn.nspname = cols.table_schema))) AS parent ON ((con.confkey[1] = parent.ordinal_position) AND (con.confrelid = parent.oid)) WHERE child.column_name != 'rowid'",
    "result": null
   }
  ],
  "previousStatements": [
   "INSERT INTO public.table_w1_124 (col112_w1_119, col112_w1_115, col112_w1_118, col112_w1_114, col112_w1_117, col112_w1_116, col112_w1_113, col88_w1_89, col88_w1_90) VALUES (e'\\x07p%^ANu'::STRING,25:::OID,e'\\x00'::STRING,true,2205:::OID,'[-8.9404324e-36,5.601245e+28,3.1780546e-34,-6.6923724e-29,-2.7986792e+37,-3.823745e+32,-2.460172e+33,-0.05216148,-5.809526e+16,3.3987968e-15,-1.5518895e+10,1.3613856e+38,-2.0548949e+06,-1.4774328e-10,-1.6587666e-36,-1.0963582e+26,7.20914e-11,-2.3527092e-13,-8.112689e-16,1.5907769e+35,2.668418e-17,2.9470206e-23,-1.3467539e+32,-1.5790973e-28,-1.1803493e-16,-1.5451226e+34,6.650306e+24,1.5215894e+10,1.0052084e-26,1.9997069e+38,2.8961443e-27,-1.7581402e-19,3.03402e+12,8.006091e+10,-44598.44,6.9682034e-15,-2.924247e-10,3.1309457e-23,-7.091033e+22,0.0013675817,3.4772007e-30,2.0948276e-07,1.6767422e+32,-9.289497e-14,-1.6791059e+27,3.8439437e+15,-2.8284505e+13,-2.4671483e-26,-2.9968157e+08,3.4092022e-28,-4.743695e-24,2.6176198e+24,-8.6685234e+11,6.471581e-19,-1.3788095e-32,-5.4894888e+17,2.9076203e-28,7.0643307e-07,0.0015376214,2.3773899e-27,-3.2007507e-27,8.174363e-07,-0.00083315815,1.6809504e-20,261148.02,1.0405816e-35,0.0013401939,-1.2516779e-33,-2.9370173e-29,-4.540797e-17,8.679558e+20,-2.0611294e-31,-2.0922858e+35,5.0715904e+08,-5.0754675e-25,-2.9700628e+14,4.5720926e-21,2.9862292e+33,-4.7747672e-32,6.2203826e+23,-6.1077554e-12,-5.2215684e-34,3.1018038e+28,-8.2912356e+13,-3.3026583e+09,-4.4792976e+33,-9809.03,-7.812911e+07,5.8832704e+15,1.354341e+24,-1.9714497e+23,-0.00010344916,-2.068134e+25,-2.9715267e+18,237.50108,-0.029341215,7.5776226e+29,-9.375345e+34,3.7711675e-23,-6.863268e-39,-1.1361602e+16,-1.4323351e-10,-2.9324616e+07,-2.9015906e-20,4.498091e-14,-1.0576723e+21,1.828886e-25,9.060454e-37,-1.868314e+15,-2.8339635e-24,-2.1948474e+24,4.006634e+30,-3.9905134e+37,-1.0914214e-06,5.2556387e-22,9.924415e-36,6.781707e-09,1.1982689e+22,1.1313396e-38,-4.4931065e+06,-8.43772e-26,5.1075355e+14,1.6865283e-10,4977.958,-2.3377064e-35,6.174903e+10,3.5412e+16,-2.3727016e+07,-5.864132e+09,-3.7331356e+07,-4.3182013e+08,-3.9564648e+30,0.05306638,9.824635e-22,-4.502427e-06,5.94575e+27,-2.137333e-14,7.8052814e-10,4.543162e-37,-1.0061144e-09,1.3390091e+28,1.480352e+16,3.5591604e-19,-1539.8691,-1.7903573e-26,-2.6291912e-06,-4.031884e+27,0.0044844216,1.5725876e-17,-9.429159e-19,4.092741e-14,4.9293683e-29,-0.1236138,-1.024087e+33,2.692711e+33,2.3087026e-20,1.326875e-16,-5.2661618e-33,2.502556e+08,-1141.8784,-1.9687636e-30,-1.039289e-27,2.3801489e+14,1.5871741e-33,2.808041e-38,-92629.555,-3.68739e-25,-1.418129e+28,0.0054536574,815.1221,236852.98,-1.7891822e-32,-5.2010405e+17,4.148726e-39,-2.3042918e+11,-9.661687e+09,-1.180768e-31,1.7654832e+13,-2.6208042e-26,-1.0454421e-26,1.2944228e+14,-1.0391547e-24,-7.99371e-12,6.978545e+34,1.1791346e+27,-2.0974784e-23,-9.810804e+27,38.117985,3.142453e-28,-1.4171715e+20,1.1521707e+35,-2.7463605e+12,-7.283175e-36,-6.17807e+26]':::VECTOR,e'\\x05'::\"char\",NULL,(-4):::INT8),(e'n\\x04u\\x1f#\\tg'::STRING,701:::OID,e'\\'w;{\\x04JXq\\x1a'::STRING,true,2950:::OID,'[1.0558952e-28,2.6941667e+21,-1.4658117e-12,-5.6172635e+30,1.4877363,-2.5555686e+26,-1.4252325e-16,-2.6085505e-12,-1.422242e-29,-2.5993517e-14,-1.4165564e-11,8.559436e-12,4.3641233e+35,-8.620481e-25,-5.809255e+14,-9.411287e+13,-2.3506209e-25,-7.993272e+22,-3.3966106e+31,9.218496e-20,4.679185e-06,2.4734896e+25,-1.105784e+33,12210.046,4.0218415e-10,2.9913223e-18,-6.986929e-39,-3.0288584e-13,0.050846476,-3.911139e+35,0.00055988284,3.370359e+11,2.096075e-07,5.269878e+20,-0.6072069,-2.818014,-2.6975e-18,10292.846,-5.6479595e+25,0.20781177,-342665.22,-6.405291e-08,1.2665696e+23,-0.0014112788,-2.2836152e+26,0.00021255358,1.24012404e+30,-0.17636393,0.00042458696,-1.907764e-19,2.6038623,-8.83681e-31,3.080833e+31,-1.4911773,-6.142173e-36,1.1180958e-09,-1.392667e+19,-1.8050116e-30,2.3377145e-35,8.7800345e-24,-0.007647992,-5.426149e+25,-1.8534506e+34,-5.0680028e+11,2.1511516e+17,-1.2223474e+06,3.6069076e+07,-2.9414233e+34,3.2520862e+07,1.2994806e-07,0.0018841962,-1846.9116,-5.869878e+17,-6.2895043e-35,-1.8684232e+21,9.324029e+13,-15.609982,1.5439235e+34,4989.4814,-1.0083593e+35,5.20232e-39,-6.512395e-24,-1.2222571e+08,-0.022930076,-3.874018e+32,4.185044e-30,-3.2552033e+19,-4.658487e-19,6.116531e-38,-5.235372e-10,426.5886,-7.146832,9.317904e+18,-1.1386374e+08,4.2065486e-14,-2.298721e-13,-3.0455818e+08,5.01767e+24,-2.5196902e+13,-1.8996206e+34,-8.076141e+07,-1.8580056e+33,-1.085828e+35,-4.244522e+32,-3.732476e-19,-7.525358e-13,1.3548649e-30,2.807966e+37,-2.815391e+23,-1.7179034e+27,0.015872562,-1.3491341e-10,4.4854117e-17,-2.7629936e+30,2.7904925e-38,-4.0162727e-28,6.7624864e-24,-10366.083,9.5407863e+24,-7.1272337e+33,-0.10469212,7.879504e-27,-2.3014096e+19,5.978476e-39,-1.2209077e-24,1.2934599e+32,-2.4496312e-29,-1.6864668e-08,-9.7147505e-29,5.7710637e-37,1.3687504e+18,-1.1750028e+24,1.7772578e-25,-1.9746987e+23,-8.000268e+08,5.6549787e-10,-5086.3906,7.970958e+34,3.9370172,2.3827007e+10,4.7274025e-11,5.8306645e-12,-2.9231569e-05,1.371384e+14,1.055312e-08,-6.783957e-22,-2.1159435e-29,-1.2920489e+09,0.0004437448,-8.517003e-27,-1.22354e-21,-1.3217425e+34,3.6416242e-29,-7.7358625e+28,-3.287852e+17,-6.9903274e-28,-2.1243403e-11,-1.5707296e+12,-2.4568627e+08,4.0971898e+29,8.54885e+32,-29535.9,3.250495e+27,-2.8388638e+29,-3.7916192e+20,2.6325093e-26,2.0423607e+36,7346.715,7.5437795e-26,-6.0549177e-34,-3.597713e+13,5.4318445e+06,4.6539514e-14,-0.049770642,-6.9918464e+16,-3.9782305e+36,3.3108274e-09,1.3308433e+35,5.8311634e-31,5.635798e-38,-1.4322201e-22,-3.0079258e-14,6.529455e-19,5.345741e+35,1.4672809e+25,-1.2732195e+11,-1.3100936e+36,-1.5934963e-31,-4.1032913e-11,2.3305178e-24,5788.5356,95.52344,-1.530749e-10,1.7760937e-33,3.4626884e+28,-1.1190459e+14,-4.7456768e-05,335.17795,-9.7135536e-11,1.7878855e-12,-0.04705731,-1.7944718e+36,-1.949422e-20,-2.6074446e+11,3.275122e-09,8.0321195e+22,-5.065618e-07,9.552565e-39,-7.389335e+31,-8.690343e-27,1.4963663,-631.858,-0.00023550636,-1.2547889e+38,8.189075e+08,3.8561026e-14,-1.2344369e-05,-1.459713e+24,-5.0621124e-16,1.817963e+15,-6.9080573e-31,4.1419844e+37,-8.6060275e-17,1.0794872e+07,-4.4455595e+06,7.508079e-24,-4.10345e-15,0.0029465482,3.0407174e-24,2.5834056e-18,-3.1731752e+18,1.5687327e+31,-1.0771906e-14,-1.9786981e+31,1.6757286e+19,0.0018541616,6.162723e-30,-1.9374032e+24,2.33671e+37,-8.784877e-30,-3.300544e-31,15.810061,-1.2027901e+36,4.3938776e+12,-2.2862018e+36,2602.5867,-1.2881371e+28,-5.980188e+13,1.0724832e+18,2.7379787e+22,-2.948031e+15,1.8036818e-26,-4.7702286e-25,-44292.074,-4.9764656e-33,-7.288491e+12,-1.0187683e-25,-0.001307965,-2.6755747e-21,1.3089352e-16,-3.0084024e+19,-0.0029549133,2.5730624e+28,7.11235e-07,-8.442219e-31,-1.6192596e+11,-3.9271278e-25,-0.07574253,1.5882605e+33,-3.3398313e-29,-3.9438483e+11,-2.724543e-14,0.10170288,1.681811e-16,10668.401,-378.95398,-2.393448e-24,-2.283212e-11,-1.4548637e-11,-3.611666e-05,-0.05839904,7.053081e+23,-0.08353069,-1.16816934e-26,-3.5662582e+15,-3.4941756e+24,2.956811e-18,-2.2934182e+35,8.2070706e-16,-1.9463326e-20,-15673.807,-4.0994223e+33,-22139.61,7.6859985e+28,-3.95081e+21,-2.4710366e+08,-4.0699335e-23,1.10375805e-17,-3.209311e-21,3.912196e+29,8.689401e-27,3.9583674e-13,-1.9714095e-20,7.4177483e-19,1.0338445e-16,-6.2263794e+34,-2.092564e-09,-3.499051e+19,9.734586e+17,-3.3793986e-16,-1.2177988e-23,2.024677e-12,2.4736832e+25,-9.1816696e-38,6.2872004e-07,-4.2674892e+15,4.4871513e-24,-4.235944e-20,0.15933676,-1.9012387e-12,2.2717735e+12,5.6905477e+13,3.1517225e-32,1.3036745e+35,3.4076182e-30,3.0007095e-13,4.8416864e-11,4.4812152e+36,2.917512e+23,-4.011681e-32,3.1341485e+28,87.32537,6.49891e+14,-5.479347e+10,-0.00017294989,-1.0464604e+24,-56.65979,-1.2871585e+38,3.068094e-39,3.8835085e+19,-5.2612533e+19,6.623312e+37,-2.5420387e+08,-2.457288e-17,2.0157032e+15,-1.9874599e-26,3.2337933e-24,-1.7614665e+34,6.975249e+24,-4.6021467e-34,4.2053445e-17,2.3945606e+11,-1.0979452e+34,5.5884137e+27,-4.34039e+17,1.4399846e-25,2.7794655e-11,5.828455e-28,2.8347584e-22,-0.0024410791,5.6767332e-27,4.6024396e+21,-2.3641372e-18,7.567021e-36,-8.1091534e+23,1.6024985e-37,1.6247844e+11,-3.1333616e-24,-1.2376372e-08,3.3668272e+18,2.0454458e+23,-1.351036e+07,-6.831511e+28,-5.7704136e-33,-1.4346192e+11,5.2925135e-19,-9130.652,7.1644174e-20,2.6197477e+32,-0.031040633,6.689686e-38,-4.3983192e+10,1.7218403e-22,-6.350345e+23,1.7466512e+13,-1.0373565e-20,1.0470108e-25,1.3767433e-22,-7.938511e-35,-1.8028234e-13,2.0558396e+26,2.1558893e+20,6.4486366e+19,2.2820754e-30,1.2610562e-14,6.5620547e+18,4.2044657e-27,3.349564e-05,-517478.84,-3.263856e+20,1.2401741e+35,-0.00011508664,-2.667556e-08,7.113652e+33,-1.2219584e+08,3.8457545e+10,-1.1428853e+32,2.3461642e-37,-1.3853027e-16,7.4316816e+28,1.9271928e+23,-2.2175885e-23,8.858258e-16,-196.2479,-2.8563809e+35,-1.6817851e+06,-1.0953672e+29,-3.9679205e-23,9.337924e+28,-2.176814e-09,2.9767385e+15,3.5612768e-25,-3.9029362e-16,-1.3235362e+11,2.7527156e-08,1.3281938e-17,-3.6374027e-37,3.7461244e-28,9.643739e+07,-5.704784e+07,6.018049e-22,-4.2866334e+11,-7.57247e-13,-1.6520112e+38,1.38359075e+14,1.9678448e-28,3.46882e+11,1.3169378e+28,8.063114e-30,-1.4950765e+06,-5.884537e+17,-3.7493677e+27,-4.266343e-16,1.5740858e+08,-4.3344625e+13,-4.2069376e-29,4.919062e+25,6.121245e+21,1.644069e+26,3.7666228e-34,597617.25,3.6223683e-36,-91487.805,1.8338295e+24,2.566656e-37,-1.2525282e-33,1.4117451e+20,2.102561e-18,-0.0037538644,-5.6741033e+15,-6.3199345e+09,1.9926472e+26,-2.4717379e-33,0.77086645,5.5261272e+17,2.2104543e-10,-3.339048e+21,6.7036813e-21,-4.6426174e-18,2.2877742e+14,2.342076e+27,-1.4747709e-28,55.50691,-6.8406997,-4.439418e+12,8.784832e-06,-0.00083627977,-1.8577588e-17,4.579214e+21,9.336972e+25,8.924296e-22,-1.2217169e+35,-9545.451,-2.4495447e-17,-1225.2518,-1.5053974e-36,-1.43088e-26,-5.95033e-16,1.01638495e+15,-9.7476164e+36,5.692231e+27,-5.028791e+33,-9.420136e-08,2.6247903e+15,3.5242928e+08,5.269894e+29,6.2716982e-24,-0.025319597,6.4482856e+13,-673.2322,-2.1409565e+30,-2.5006648e-35,-8.3602815e+06,3.3205263e-30,1.1388457e+33,1.7972944e+15,3.8463693e+20,-0.00035653063,1.1097119e+17,-1.6017943e+12,5.28449e+36,8.804614e-08,5.425825e+36,-238759,4.191295e+22,-1.6067075e-21,3.0585207e+37,1.6493811e-31,4.6967677e-37,-1.7742328e+22,2.3311824e+08,4.157782e+36,-5.0678666e-33,2.4040527e+21,7.4382036e+20,9.140407e+16,1.4069694e+13,8.0533055e-22,2.4527407e-12,-2.7124554e-20,-0.005991269,3.7642842e+22,-1.0435765e-16,1.8354389e+33,-5.527625e-14,-1.9272203e-13,9.002224e+34,-3.3024463e+13,4.6973583e+17,3.4551407e+31,-0.09780094,498111.44,5.3414833e+19,7.5272154e+18,-8.170128e+37,-4.5110778e+11,-0.0002765098,-1.9342514e-21,6.2565807e+16,3164.9746,2.2187317e-15,2.5206433e-23,1.2151532e+11,-8.4130326e-13,-3.859729e-14,-4.0642383e+09,2.8042947e+32,5.4167768e+11,-0.013234823,-3.4010317e-15,-2.080173e+23,-1.05066824e-35,-9.9546034e-33,-9.549704e-14,-5.4270682e-33,-1.0085333e+37,-6.7139735e+35,1.1277995e+31,3.4469405e+34,1.9534247e-18,-1.1224656e-36,5.863614e+17,7636.226,-2.803778e+33,-6.7614653e-19,3.2246756e+07,-1.2618268e-37,1.1694671e-22,-0.15245113,1.1292199e+33,5.933473e+32,5.7554322e-11,1.2164743e-25,-1.1921016e+13,2.1675718e+08,-8.636424e-34,-6.674431e+15,5.387921e+07,-6.5615783e+24,-3085.581,7.737287e+17,0.0008676976,-1.20728085e-08,-5.2455703e+10,-2.264514e+22,0.0005442248,7.414669e-18,3.818047e-18,-6.908932e-38,4.4069574e+26,-4.3885234e-14,-5.295597e+28,-3.0283115e-25,3.5867274e-21,1.08328e+11,1.1607103e+25,2.0691441e-07,3.8854972e+22,-8.2822705e+15,1.1833512e-37,-4.5051155e+25,-1.7150059e-16,1.4659924e-31,1.23031965e+29,2.4625003e-15,6.117329e+35,-3.3450283e+10,-1.2134042e+12,2.8737757e-12,-3.1479554e+29,-2.2197568e+09,-2.6867646e+38,6.2030025e+27,1.9683482e-24,3.6578934e+30,7.4733e-11,-1.9806485e+14,-6.649875e+11,2.7139077e+17,1.8877959e+18,5.82619e-28,0.1328821,0.0036019818,-0.27224246,-3.4180478e-19,-1.3042597e-21,0.7019636,0.00038273347,5.5785517e-06,-1.17283484e-07,3.11991e+14,5.426232e+30,1.1932813e-38,-2.9285975e-12,-1.0209311e-06,2.0183653e-11,3.242928e+11,-1.3462883e-35,-9.1320925e-35,-1.8334551e-38,2.2154094e-32,-7.891697e-11,5.6553727e-11,-1.8336504,-2.2477892e+10,1.2009661e+13,-1.7431383e-34,-6.6569624e-26,1.5016365e-12,-4.719529e-33,-3.2272337e+16,8.2303236e-32,-3.21796e+16,3.8555885e+06,-6.35786e+28,2.5538613e+23,3.1499044e-15,-1.6364558e+22,478555.78,1.0709054e-31,2.392854e-32,1.6283491e-16,-5.78238e-13,-6421.4937,-4.84967e-09,-1.470497e+38,-1.3279742e-12,3.3709107e-34,9.941091e-26,-9.8061074e+10,-2.3938753e-33,-3.5251948e-34,-9838.957,4.3576055e-15,4.69452e-28,7.507101e+26,-0.027296174,-3.860365e+35,1.8560388e+21,-0.018810201,1.3838457e+14,5.184727e+22,-1.4949616,0.00013264,-2.7409389e-05,-3.7607907e-15,-9.876845e-38,1.9648663e+33,-1.6063951e+22,-5.738337e+20,-7.1559213e+18,6.2212704e+32,2.221195e+33,-3.0088505e+25,0.0031193323,-2.5801674e-38,-4.5982927e+33,1.623201e-17,5.316393e-34,5.2645812e+07,7.733875e+21,-4.6541734e-10,1.7534707e+32,-1.3449443e+26,1.419497e+26,-1.7596131e-31,0.0003809806,1.0079425e-27,0.91271853,-2.978088e-30,2.4383347e+13,7.733406e+26,-2.57353e-23,6.857954e+20,-6.161975e+06,1.2749635e-09,1.7428148e+19,-4.158969,-1.5629188e-21,4.6434727,2.9848283e-18,7.685396e+11,9.9547494e+08,2.6051826e+19,4.8094086e-21,4.0457845e-23,68.86047,-0.0010785764,3.8105007e-25,-9.412597e+34,1.9339399e-36,5.794144e+29,-1.771816e-34,1.6893573e-08,1.7003413e-32,-3.2031743e-37,2.2155542e-28,3.0727045e-37,8.2838164e+21,4.1285382e+30,-4.2784868e+35,-7.009262e-26,8.708053e-35,-3.1468849e-12,-219401.75,3.305131e-05,2.421742e+20,3.385012e+36,-8.602484e-17,-6.68311e-26,4.6400194e-07,-9.870506e-30,1.7039894e-22,-1.14554e+07,-3.3171086e-38,-1.5293338e+06,-417.29547,2.2363489e-05,-2.9090426e-35,-1.0273865e-11,3.4099234e-05,-3.002346e+15,6.0679777e+27,-3.074202e+07,-1.0717009e+20,-3.358898e-25,5.9632976e+07,-1.3124021e-17,-2.3920534e-32,2.7337705e+33,1.1296554e-05,-8.38922e+09,-2.5688366e-11,-1.5703168e-11,-4.6187182e-33,-2.13949e+06,5.432197e-39,8008.4385,-2.6342038e-20,-1.233892e-34,1.0929451e+27,-3.3022313e-06,447761.88,9.090244e-34,3.218069e-05,9.483162e+35,-5.944747e-15,-1.1933271e-20,2.880533e-36,2.3409802e-16,-1.161091e-22,5.1630235e+32,4.416812e+31,-1.8503731e-14,1.7028315e+12,-4.7870448e-18,4.384454e+23,-4.6416717e-16,2.5179758e-24,-1.2607793e-14,-2.1899432e-30,-3.9779354e+15,8.250641e+30,897993.56,5.582985e+12,-1.002443e+31,0.17714426,-3.8778605e-25,1.0768461e+31,-3.4236604e-22,-7.1729806e-14,1.8837915e+28,-4.3050921e+18,5.2285545e-38,-9.9077695e+33,2.2038672e-36,4.2321094e-06,2.4697541e+08,3.384437e-23,-8.520888e-21,-8.445017e-35,-4.1967983,-1.208189e-30,-5.017419,-5.041896e-18,-3.9232972e+07,2.7498945e+24,2.287682e+29,-1.0590261e-17,3.2134556e+23,-8.311788e+14,5.907191e+17,-1.1673083e-27,1658.6315,-251.26768,-3.5436996e-35,3.58752e-11,-7.487529e-31,2.449101e+37,8.735973e+28,-0.00047352447,-0.0039477856,1.0663846e+22,5.0472096e-36,2.103212e+31,0.00044060952,8.5449674e-16,-2.5692256e-21,5.182764e+29,3.1623704e-27,4.9992543e+30,-3.0806653e-23,-1.331327e+31,-1.8114641e+32,5.152954e+09,-1.1070275e+33,-936.7972,-2.6407505e+32,-2.761909e-28,4.4812085e+14,9.7159005e-30,0.100133196,2.886234,-5.634055e-12,6.657801e-31,-2.0105778e-11,5.6129743e-17,4.513034e-28,3.4555395e+30,-9.511465e-33,1.2374309e-19,-7121.485,-1.5152551e-37,6.1229125e+11,-0.00036285215,6.168763e-15,-2.370692e+22,-5.948418e+20,-4.0501814e-30,2.3105464e+24,1.4006467e-35,-1.6120593e+15,-1.1016774e+30,-8.139192e+34,-7.113459e-27,-6.2024437e-06,8.276923e-05,4.136565e-16,-5.95847e-21,73.588745,1.556385,6.723293e-13,-3.6017302e-12,2.4404894e+20,-1.9575544e+15,1.5406885e-10,-4.5798205e-07,-83932.38,-2.1427861e+18,4.6048144e+16,9.546136e+16,-1.0671187e-11,9.4898375e-34,2.7033898e+11,1.9643627e+18,1.5557532e+11,2.051716e+09,6.755796e+15,-0.011395583,2.0630135e-34,-0.009493616,1.3104368e+16,1.6008195e-17,5.1501245e+08,1.5076444e+27,5.8148133e+26,7.446646e-17,-9.0162367e-13,-8.794197e-28,1.1718591e-29,0.00084520713,-2.2069224e-07,2.2418832e-33,-5.2190925e-34,13.735793,-2.578696e+13,2.7377045e+18,-2.996963e-14,3.2797723e+38,-1.2658737e-06,1.1683012e-37,-1.8429111e+31,-2.2977165e+08,2.747433e-17,-6.9692917e-13,3.5979543e-16,-3.489444e-28,1.3671846e+37,5.6246677e+34,-1.3832865e-12,3.3933237e+28,2.9689692e+23,-1.3083475e-18,-6.190345e-39,9.566384e-11,-3.3093212e-10,3.1994668e-06,-0.24438871,-4.3016385e-07,3.119651e+24,1.0780013,-4.3884823e+11,-5.6471792e-36,4.7131553e+33,-6.606379e+16,-5.4211614e+31,-2.6057796e+07,2.4843635e-32,3.922192e-17,2.3540164e+36,-1.74883e+07,-1.5567686e+22,1.614801e-09,2.857377e-30,3.90024e-14,133125.61,-8.4596423e-19,5.731239e+07,-4.5205173e+23,-1.5166675e-11,-1.282659e+16,4.4301863e+27]':::VECTOR,e'\\x19'::\"char\",'''kKganTQd''':::TSQUERY,(-119):::INT8)"
  ],
  "usesDeclarativeSchemaChanger": true
 }
}
    schema_change_external_test.go:164: 
        	Error Trace:	pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:164
        	Error:      	Received unexpected error:
        	            	***UNEXPECTED ERROR; Received an unexpected execution error.: ERROR: insert on table "table_w1_124" violates foreign key constraint "table_w1_112_col112_w1_119_table_w1_124_col112_w1_119_fk" (SQLSTATE 23503)
        	            	(1) forced error mark
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*opStmt).executeStmt
        	            	  | 	pkg/workload/schemachange/operation_generator.go:3273
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	pkg/workload/schemachange/schemachange.go:511
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	pkg/workload/schemachange/schemachange.go:599
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func5.func7
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:153
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	external/org_golang_x_sync/errgroup/errgroup.go:78
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (4) ***UNEXPECTED ERROR; Received an unexpected execution error.
        	            	Wraps: (5) ERROR: insert on table "table_w1_124" violates foreign key constraint "table_w1_112_col112_w1_119_table_w1_124_col112_w1_119_fk" (SQLSTATE 23503)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *pgconn.PgError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 130
Total Schema Statements Succeeded = 130
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 90
Total Transactions Rolled Back = 85
Total Transactions Executed = 175
    schema_change_external_test.go:117: backup, tracing data, and system table dumps in /mnt/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/testlogs/pkg/ccl/testccl/workload/schemachange/schemachange_test/run_5_of_10/test.outputs/logTestWorkload879047053
    schema_change_external_test.go:126: running DROP with use_declarative_schema_changer = on
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestWorkload879047053
--- FAIL: TestWorkload (13.51s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=5</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #154221 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [depends-on relation is dropped during DROP SCHEMA CASCADE] [C-test-failure O-robot T-sql-foundations branch-release-25.3]
- #153823 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [did not received expected DependentObjectsStillExist error] [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.3]
- #153126 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [column &#34;crdb_internal_column_3_name_placeholder&#34; does not exist] [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.3.20-rc]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-55089

--- a/pkg/workload/schemachange/error_screening.go
+++ b/pkg/workload/schemachange/error_screening.go
@@ -8,7 +8,6 @@ package schemachange
 import (
 	"context"
 	"fmt"
-	"regexp"
 	"slices"
 	"strings"
 
@@ -789,36 +788,6 @@ func (og *operationGenerator) scanStringArrayNullableRows(
 	return results, nil
 }
 
-func (og *operationGenerator) scanStringArrayRows(
-	ctx context.Context, tx pgx.Tx, query string, args ...interface{},
-) ([][]string, error) {
-	rows, err := tx.Query(ctx, query, args...)
-	if err != nil {
-		return nil, errors.Wrapf(err, "scanStringArrayRows: %q %q", query, args)
-	}
-	defer rows.Close()
-
-	var results [][]string
-	for rows.Next() {
-		var columnNames []string
-		err := rows.Scan(&columnNames)
-		if err != nil {
-			return nil, errors.Wrapf(err, "scan: %q, args %q, scanArgs %q", query, columnNames, args)
-		}
-		results = append(results, columnNames)
-	}
-
-	if rows.Err() != nil {
-		return nil, rows.Err()
-	}
-
-	og.LogQueryResults(
-		query,
-		results,
-		args...)
-	return results, nil
-}
-
 func (og *operationGenerator) indexExists(
 	ctx context.Context, tx pgx.Tx, tableName *tree.TableName, indexName string,
 ) (bool, error) {
@@ -1049,188 +1018,6 @@ func (og *operationGenerator) constraintExists(
 	return og.scanBool(ctx, tx, `SELECT EXISTS(
 		SELECT * FROM information_schema.table_constraints WHERE table_name = $1 AND constraint_name = $2
 	 )`, string(tableName), string(constraintName))
-}
-
-var (
-	// regexpUnknownSchemaErr matches unknown schema errors with
-	// a descriptor ID, which will have the form: unknown schema "[123]"
-	regexpUnknownSchemaErr = regexp.MustCompile(`unknown schema "\[\d+]"`)
-)
-
-// checkAndAdjustForUnknownSchemaErrors in certain contexts we will attempt to
-// bind descriptors without leasing them, since we are using crdb_internal tables,
-// so it's possible for said descriptor to be dropped before we bind it. This
-// method will allow for "unknown schema [xx]" in those contexts.
-func (og *operationGenerator) checkAndAdjustForUnknownSchemaErrors(err error) error {
-	if pgErr := new(pgconn.PgError); errors.As(err, &pgErr) &&
-		pgcode.MakeCode(pgErr.Code) == pgcode.InvalidSchemaName {
-		if regexpUnknownSchemaErr.MatchString(pgErr.Message) {
-			og.LogMessage(fmt.Sprintf("Rolling back due to unknown schema error %v",
-				err))
-			// Force a rollback and log inside the operation generator.
-			return errors.Mark(err, errRunInTxnRbkSentinel)
-		}
-	}
-	return err
-}
-
-// violatesFkConstraints checks if the rows to be inserted will result in a foreign key violation.
-func (og *operationGenerator) violatesFkConstraints(
-	ctx context.Context,
-	tx pgx.Tx,
-	tableName *tree.TableName,
-	nonGeneratedColNames []tree.Name,
-	rows [][]string,
-) (expectedViolation, potentialViolation bool, err error) {
-	// TODO(annie): readd the join on active constraints once #120702 is resolved.
-	//
-	// N.B. We add random noise to column names that makes it hard to just directly call on these names. This is
-	// not the case with table/schema names; thus, only column names are quote_ident'ed to ensure that they get
-	// referenced properly.
-	fkConstraints, err := og.scanStringArrayRows(ctx, tx, fmt.Sprintf(`
-		  SELECT array[
-			  parent.table_schema,
-			  parent.table_name,
-			  parent.column_name,
-			  parent.is_generated,
-			  child.column_name,
-			  child.is_generated
-		  ]
-		  FROM (
-		        SELECT conname, conkey, confkey, conrelid, confrelid
-		          FROM pg_constraint
-		         WHERE contype = 'f'
-		           AND conrelid = '%s'::REGCLASS::INT8
-		       ) AS con
-		  JOIN (
-		        SELECT column_name, ordinal_position, column_default, is_generated
-		          FROM information_schema.columns
-		         WHERE table_schema = '%s'
-		           AND table_name = '%s'
-		       ) AS child ON conkey[1] = child.ordinal_position
-		  JOIN (
-		        SELECT pc.oid,
-		               cols.table_schema,
-		               cols.table_name,
-		               cols.column_name,
-		               cols.ordinal_position,
-		               cols.is_generated
-		          FROM pg_class AS pc
-		          JOIN pg_namespace AS pn ON pc.relnamespace = pn.oid
-		          JOIN information_schema.columns AS cols ON (pc.relname = cols.table_name AND pn.nspname = cols.table_schema)
-		       ) AS parent ON (
-		                       con.confkey[1] = parent.ordinal_position
-		                       AND con.confrelid = parent.oid
-		                      )
-		 WHERE child.column_name != 'rowid';
-`, tableName.String(), tableName.Schema(), tableName.Object()))
-	if err != nil {
-		return false, false, og.checkAndAdjustForUnknownSchemaErrors(err)
-	}
-
-	// Maps a non-generated column name to its index. This way, the value of a
-	// column in a row can be looked up using row[colToIndexMap["columnName"]] = "valueForColumn"
-	nonGeneratedColumnNameToIndexMap := map[tree.Name]int{}
-
-	for i, name := range nonGeneratedColNames {
-		nonGeneratedColumnNameToIndexMap[name] = i
-	}
-
-	for _, row := range rows {
-		for _, constraint := range fkConstraints {
-			parentTableSchema := tree.Name(constraint[0])
-			parentTableName := tree.Name(constraint[1])
-			parentColumnName := tree.Name(constraint[2])
-			parentIsGenerated := constraint[3] == "ALWAYS"
-			childColumnName := tree.Name(constraint[4])
-			childIsGenerated := constraint[5] == "ALWAYS"
-
-			// If self referential, there cannot be a violation.
-			parentAndChildAreSame := parentTableSchema == tableName.SchemaName && parentTableName == tableName.ObjectName
-			if parentAndChildAreSame && parentColumnName == childColumnName {
-				continue
-			}
-
-			// If the foreign key involves any generated columns, skip detailed FK checking
-			// and conservatively assume a violation may occur. This avoids the complexity
-			// of reasoning about values that are automatically computed by the database.
-			if parentIsGenerated || childIsGenerated {
-				return false, true, nil
-			}
-
-			violation, err := og.violatesFkConstraintsHelper(
-				ctx, tx, nonGeneratedColumnNameToIndexMap, parentTableSchema, parentTableName, parentColumnName, childColumnName, tableName, parentAndChildAreSame, row, rows,
-			)
-			if err != nil {
-				return false, false, err
-			}
-			if violation {
-				return true, false, nil
-			}
-		}
-	}
-
-	return false, false, nil
-}
-
-// violatesFkConstraintsHelper checks if inserting a single row would violate a
-// foreign key constraint between the given child and parent columns.
-//
-// This function assumes that neither the parent nor child columns are generated.
-// The caller must ensure this; generated columns are not supported.
-func (og *operationGenerator) violatesFkConstraintsHelper(
-	ctx context.Context,
-	tx pgx.Tx,
-	nonGeneratedColumnNameToIndexMap map[tree.Name]int,
-	parentTableSchema, parentTableName, parentColumn, childColumn tree.Name,
-	childTableName *tree.TableName,
-	parentAndChildAreSameTable bool,
-	rowToInsert []string,
-	allRows [][]string,
-) (bool, error) {
-
-	childIndex, ok := nonGeneratedColumnNameToIndexMap[childColumn]
-	if !ok {
-		return false, errors.Newf("child column %s does not exist in table %s", childColumn, childTableName)
-	}
-	childValue := rowToInsert[childIndex]
-	// If the value to insert in the child column is NULL and the column default is NULL, then it is not possible to have a fk violation.
-	if childValue == "NULL" {
-		return false, nil
-	}
-	// If the parent and child are the same table, then any rows in an existing
-	// insert may satisfy the same constraint.
-	var parentAndChildSameQueryColumns []string
-	if parentAndChildAreSameTable {
-		for _, otherRow := range allRows {
-			parentIdx, ok := nonGeneratedColumnNameToIndexMap[parentColumn]
-			if !ok {
-				return false, errors.Newf("parent column %s does not exist in table %s", parentColumn, childTableName)
-			}
-			parentValueInSameInsert := otherRow[parentIdx]
-
-			// Skip over NULL values.
-			if parentValueInSameInsert == "NULL" {
-				continue
-			}
-			parentAndChildSameQueryColumns = append(parentAndChildSameQueryColumns,
-				fmt.Sprintf("%s = %s", parentValueInSameInsert, childValue))
-		}
-	}
-	checkSharedParentChildRows := ""
-	if len(parentAndChildSameQueryColumns) > 0 {
-		// Check if none of the rows being inserted satisfy the foreign key constraint,
-		// since the foreign key constraints refers to the same table. So, anything in
-		// the insert batch can satisfy the constraint.
-		checkSharedParentChildRows = fmt.Sprintf("NOT (true = ANY (ARRAY [%s])) AND",
-			strings.Join(parentAndChildSameQueryColumns, ","))
-	}
-	q := fmt.Sprintf(`
-	    SELECT %s count(*) = 0 from %s.%s
-	    WHERE %s = (%s)
-	`,
-		checkSharedParentChildRows, parentTableSchema.String(), parentTableName.String(), parentColumn.String(), childValue)
-	return og.scanBool(ctx, tx, q)
 }
 
 func (og *operationGenerator) columnIsInAddingOrDroppingIndex(
@@ -1763,35 +1550,6 @@ func (og *operationGenerator) tableHasUniqueConstraintMutation(
 			FROM table_desc)
 			WHERE (m->>'direction')::STRING IN ('ADD', 'DROP')
 			AND (m->'index'->>'unique')::BOOL IS TRUE
-		);`, tableName)
-}
-
-// tableHasForeignKeyMutation determines if a table has any foreign key constraint
-// mutation ongoing. This means either being added or dropped.
-func (og *operationGenerator) tableHasForeignKeyMutation(
-	ctx context.Context, tx pgx.Tx, tableName *tree.TableName,
-) (bool, error) {
-	return og.scanBool(ctx, tx, `
-		WITH table_desc AS (
-			SELECT crdb_internal.pb_to_json(
-				'desc',
-				descriptor,
-				false
-			)->'table' as d
-			FROM system.descriptor
-			WHERE id = $1::REGCLASS
-		)
-		SELECT EXISTS (
-			SELECT * FROM (
-			SELECT jsonb_array_elements(
-				CASE WHEN d->'mutations' IS NULL
-				THEN '[]'::JSONB
-				ELSE d->'mutations'
-				END
-			) as m
-			FROM table_desc)
-			WHERE (m->>'direction')::STRING IN ('ADD', 'DROP')
-			AND (m->'constraint'->>'foreign_key') IS NOT NULL
 		);`, tableName)
 }
 

--- a/pkg/workload/schemachange/operation_generator.go
+++ b/pkg/workload/schemachange/operation_generator.go
@@ -3134,9 +3134,6 @@ func (og *operationGenerator) insertRow(ctx context.Context, tx pgx.Tx) (stmt *o
 	// we will need to evaluate generated expressions below.
 	hasUniqueConstraints := false
 	hasUniqueConstraintsMutations := false
-	hasForeignKeyMutations := false
-	expectedFkViolation := false
-	potentialFkViolation := false
 	if !anyInvalidInserts {
 		// Verify if the new row may violate unique constraints by checking the
 		// constraints in the database.
@@ -3151,31 +3148,17 @@ func (og *operationGenerator) insertRow(ctx context.Context, tx pgx.Tx) (stmt *o
 		if err != nil {
 			return nil, err
 		}
-		// Verify if the new row will violate fk constraints by checking the constraints and rows
-		// in the database.
-		expectedFkViolation, potentialFkViolation, err = og.violatesFkConstraints(ctx, tx, tableName, nonGeneratedColNames, rows)
-		if err != nil {
-			return nil, err
-		}
-		// Determine if a foreign key mutation exists.
-		hasForeignKeyMutations, err = og.tableHasForeignKeyMutation(ctx, tx, tableName)
-		if err != nil {
-			return nil, err
-		}
 	}
 
 	stmt.potentialExecErrors.addAll(codesWithConditions{
 		{code: pgcode.UniqueViolation, condition: hasUniqueConstraints || hasUniqueConstraintsMutations},
-		{code: pgcode.ForeignKeyViolation, condition: expectedFkViolation || potentialFkViolation || hasForeignKeyMutations},
+		{code: pgcode.ForeignKeyViolation, condition: true},
 		{code: pgcode.NotNullViolation, condition: true},
 		{code: pgcode.CheckViolation, condition: true},
 		{code: pgcode.InsufficientPrivilege, condition: true}, // For RLS violations
 	})
-	og.expectedCommitErrors.addAll(codesWithConditions{
-		{code: pgcode.ForeignKeyViolation, condition: expectedFkViolation && !hasForeignKeyMutations},
-	})
 	og.potentialCommitErrors.addAll(codesWithConditions{
-		{code: pgcode.ForeignKeyViolation, condition: potentialFkViolation || hasForeignKeyMutations},
+		{code: pgcode.ForeignKeyViolation, condition: true},
 	})
 
 	var formattedRows []string

